app-backend/http4s-util/src/main/scala/com/rasterfoundry/http4s/xray/XRayTracingContext.scala

```diff
@@ -0,0 +1,117 @@
+package com.rasterfoundry.http4s.xray
+
+import java.time.Instant
+
+import cats.data._
+import cats.implicits._
+import cats.effect.concurrent.Ref
+import cats.effect.{Resource, Sync, Timer}
+import com.amazonaws.services.xray.AWSXRayAsync
+import com.amazonaws.xray.entities.TraceID
+import com.colisweb.tracing.TracingContext
+import com.colisweb.tracing.TracingContext.TracingContextResource
+
+import scala.util.Random
+
+class XRayTracingContext[F[_]: Sync: Timer](
+    client: AWSXRayAsync,
+    val segment: Segment[F],
+    val tagsRef: Ref[F, Map[String, String]])
```

Comment:
Lifted the approach of being able to update tags from an update on scala-opentracing -- should probably update the library as part of this PR too